### DISCUSSION TITLE
OpenCollective and MAUI OpenGL.Control - .NET 7/8+

### DISCUSSION THREAD

**Discussion by najak3d** [opentk/opentk#1629] [Ideas]
We have financial need for OpenGL support in MAUI.    We currently employ OpenTK on .NET 4.7 and wanting to make the leap to .NET 7/8.... where currently there isn't OpenTK support.   We aren't really a MAUI app (we have our own Graphics-based UI), but .NET 7/8 generally fall under the "umbrella" term MAUI now.   

Here is the link to the page where they've announced OpenGLView is obsolete for .NET 7+:
https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.openglview?view=net-maui-7.0&viewFallbackFrom=net-maui-6.0

We need this control, or an equivalent replacement, to make this leap to MAUI .NET 7/8+.

===
I firmly believe all good Open Source Projects should have OpenCollective accounts, to allow for defining "objectives with bounties" so that those (like us) with business needs for certain objectives, can "give back" to those who are supplying these awesome solutions for the world.   This works out as a "win-win-win" - businesses win because they can help accelerate development of stuff they need, while the developers win because it's rightful to get paid for your work (and may even allow you to work more on open source), and the world wins, as now everyone else can also benefit from the accelerated progress.

**Comment by NogginBops** [comment]
A MAUI control has been on the todo list for quite some time.

If we are to put a bounty on this task I think we should discuss what the requirements should be put on the task.
What platforms should the OpenGL view support? Should there be any other features implemented?

I'm not very familiar with Open Collective and how it operates, but I could probably open one for OpenTK.
Would GitHub Sponsors be an alternative or is Open Collective your only option for this?

**Reply by najak3d** [reply]
Both methods should work fine for us.  I've only had experience with OpenCollective so far.

We won't *need* this until April 2024, when we plan to start our work, and can delay our needs until August 2024.   As time passes, the value of this component goes up.   So for starters, we can fund at $200, to get the bounty established/started, and then if others don't jump aboard to help fund it, then we can then increase our contribution from there.

At this point, it's worth ~$200 to us to get it done sooner-than-later, to remove this risk next year.   As time progresses, the value of this component goes up for us.  Our belief is that "we're not alone" on needing/wanting it -- and so the existence of a "way to contribute" might obtain enough contributors to fuel the effort.

Our requirements are simple -- make it work the previous OpenGL components from Xamarin.   Each application should be able to run multiple instances of this control, and navigating to a new form, shouldn't auto-dispose of the OpenGL context -- so that when you go back to the form with OpenGL control, it doesn't have to "fully reinitialize".   (UrhoSharp's Xamarin Control behaved like that, which was awful to deal with.)

**Comment by BuslikDrev** [comment]
Какую версию OpenTK в .NET 4.7 используете?
